plugins/modules/sonic_mgmt_servers.py

```diff
@@ -0,0 +1,496 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2024 Dell Inc. or its subsidiaries. All Rights Reserved
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#############################################
+#                WARNING                    #
+#############################################
+#
+# This file is auto generated by the resource
+#   module builder playbook.
+#
+# Do not edit this file manually.
+#
+# Changes to this file will be over written
+#   by the resource module builder.
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for sonic_mgmt_servers
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+module: sonic_mgmt_servers
+version_added: 2.5.0
+notes:
+- Tested against Enterprise SONiC Distribution by Dell Technologies.
+- Supports C(check_mode).
+short_description: Manage management servers configuration on SONiC
+description:
+  - This module provides configuration management of management servers for devices running SONiC
+author: Shade Talabi (@stalabi1)
+options:
+  config:
+    description:
+      - Management servers configuration
+    type: dict
+    suboptions:
+      rest:
+        description:
+          REST server configuration
+        type: dict
+        suboptions:
+          api_timeout:
+            description:
+              - Maximum time in seconds the REST server will wait for a REST API request-response cycle to complete
+              - Range 0-4294967295
+            type: int
+            default: 900
+          client_auth:
+            description:
+              - Client authentication methods list
+              - Specify as a comma separated list. Options for list are password, jwt, cert, and none.
+            type: str
```

Comment:
After discussing this suggestion, the conclusion is that, since multiple values can be specified, adding a 'choices' attribute would require this option to become a 'list' and that would add undesirable complexity to the implementation and to the user interface.

It is sufficient to provide the current description in this file, along with a statement of the default value that will help to illustrate the possible user input for this option.

For these reasons, I will mark this issue "Resolved".